### PR TITLE
Add terraform plan for heat

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Add terraform plan for heat service.
Add enable-heat variable to decide on deploy or
destroy heat application.

### CODE REVIEW

**Review by gboutry — CHANGES_REQUESTED** [review]
I believe all this should be added to your plugin plan, and not the openstack plan

**Comment by hemanthnakkina** [comment]
> I believe all this should be added to your plugin plan, and not the openstack plan

Its a choice, being in sunbeam-terraform plan for openstack components will have advantage
 - cluster operations like resize need not wait on each plugin resize completion
 - testing for openstack components just with terrform plans will be in single place

**Review by gboutry — APPROVED** [review]
Ok, I understand, we'll really need to look into having a dedicated runner for the testing, adding resources, will make it fail even more so.

File: main.tf [hunk 1]
```diff
@@ -319,3 +319,60 @@ resource "juju_offer" "ca-offer" {
...
+module "mysql-heat-cfn" {
+  count      = var.enable-heat ? (var.many-mysql ? 1 : 0) : 0
+  source     = "./modules/mysql"
+  model      = juju_model.sunbeam.name
+  name       = "mysql"
+  channel    = var.mysql-channel
+  scale      = var.ha-scale
+  many-mysql = var.many-mysql
+  services   = ["heat"]
+}
```
Comment (javacruft) [inline main.tf:359]: I believe we'll just want a single mysql-heat to support all heat-* services - they will share common state - its just a different API per service.

Comment (hemanthnakkina) [reply main.tf]: Fixed.

**Review by javacruft — APPROVED** [review]

**Pull request merged**